src/blocks/custom/checkbox/checkbox.php

```diff
@@ -9,17 +9,18 @@
 
 use Eightshift_Libs\Helpers\Components;
 
-$block_class = $attributes['blockClass'] ?? '';
-$name        = $attributes['name'] ?? '';
-$value       = $attributes['value'] ?? '';
-$label       = $attributes['label'] ?? '';
-$checkbox_id = $attributes['id'] ?? '';
-$classes     = $attributes['classes'] ?? '';
-$theme       = $attributes['theme'] ?? '';
-$is_checked  = isset( $attributes['isChecked'] ) && $attributes['isChecked'] ? 'checked' : '';
-$is_disabled = isset( $attributes['isDisabled'] ) && $attributes['isDisabled'] ? 'disabled' : '';
-$is_read_only = isset( $attributes['isReadOnly'] ) && $attributes['isReadOnly'] ? 'readonly' : '';
-$is_required = isset( $attributes['isRequired'] ) && $attributes['isRequired'] ? 'required' : '';
+$block_class     = $attributes['blockClass'] ?? '';
+$name            = $attributes['name'] ?? '';
+$value           = $attributes['value'] ?? '';
+$label           = $attributes['label'] ?? '';
+$checkbox_id     = $attributes['id'] ?? '';
+$classes         = $attributes['classes'] ?? '';
+$theme           = $attributes['theme'] ?? '';
+$is_checked      = isset( $attributes['isChecked'] ) && $attributes['isChecked'] ? 'checked' : '';
+$is_disabled     = isset( $attributes['isDisabled'] ) && $attributes['isDisabled'] ? 'disabled' : '';
+$is_read_only    = isset( $attributes['isReadOnly'] ) && $attributes['isReadOnly'] ? 'readonly' : '';
+$is_required     = isset( $attributes['isRequired'] ) && $attributes['isRequired'] ? 'required' : '';
```

Comment:
Could WP functions be used here?

https://developer.wordpress.org/reference/functions/checked/
https://developer.wordpress.org/reference/functions/selected/
https://developer.wordpress.org/reference/functions/disabled/
https://developer.wordpress.org/reference/functions/readonly/

?

It could be overkill but could be cool to use 🤷‍♂️  Thoughts?

Reply:
I don't think they can. As it states they output the attributes and should be used inside form elements.
These variable should contain boolean values and used as arguments to these function listed above.

Reply:
The `$is_checked` variable does a boolean check with `isset` and what `$attributes['isChecked']` inside the ternary. The point is that you'd just have the booleans in these variables, and down in the code you'd use the function with the boolean provided.

Looking at it now, I think it might be an overkill.

Reply:
Yeah in attributes you do have a boolean so we could in theory use this.

I personally find them confusing a bit, especially in this situation ("checking checked vs current" what is checked vs what is current in this case?).

Looking at the docs / example, it seems it makes most sense when you're in a form's controller (you have the current saved value vs what you received in the form. Then you compare the two to figure out of checkbox should be checked or not). 

In our case, we only really have 1 value (the value that is passed to the component) so I think it makes more sense how it is currently.